### PR TITLE
fix: show sat bundle for each input with an exotic or inscription

### PR DESCRIPTION
**Pull request by fedeerbes**
# 🔘 PR Type

- Bugfix

# 📜 Background
This PR fixes an UI issue where we were mixing all the inputs sat bundles under one bundle. The fix shows a sat bundle for each input with an exotic or inscription and it also removes the bundle size component

Issue Link: #[ENG-3392](https://linear.app/xverseapp/issue/ENG-3392/%5Bsatr-ext-12%5D-see-sent-and-received-rare-sats-in-psbt-signing-tx)

# 🔄 Changes
- Change logic in useDetectOrdinalInSignPsbt.ts to have an array of bundles
- Use changes in useDetectOrdinalInSignPsbt.ts to show a sat bundle for each input with an exotic or inscription with the input number as a title in single PSBT and batch PSBT
- Remove bundle size component

# ✅ Review checklist
Please ensure the following are true before merging:

- [ ] Code Style is consistent with the project guidelines.
- [ ] Code is readable and well-commented.
- [ ] No unnecessary or debugging code has been added.
- [ ] Security considerations have been taken into account.
- [ ] The change has been manually tested and works as expected.
- [ ] Breaking changes and their impacts have been considered and documented.
- [ ] Code does not introduce new technical debt or issues.

### CODE REVIEW

File: src/app/screens/signPsbtRequest/index.tsx [hunk 1]
```diff
@@ -387,12 +387,21 @@ function SignPsbtRequest() {
...
-              {bundleItemsData && (
-                <SatsBundle
-                  title={userReceivesOrdinal ? t('YOU_WILL_RECEIVE') : t('YOU_WILL_TRANSFER')}
-                  bundle={bundleItemsData as Bundle}
-                />
-              )}
+              {Array.isArray(bundleItemsData) &&
+                bundleItemsData.map((bundle, index) => (
+                  <SatsBundle
+                    title={`${tCommon('INPUT')} #${bundle.inputIndex + 1}`}
```
Comment (abdulhaseeb4239) [inline src/app/screens/signPsbtRequest/index.tsx:393]: we need to decide whether we want to add +1 here

File: src/app/hooks/useDetectOrdinalInSignPsbt.ts [hunk 2]
```diff
@@ -1,33 +1,41 @@
...
       const inputsResponse = await Promise.all(inputsRequest);
-      inputsResponse.forEach((inputResponse) => {
+      inputsResponse.forEach((inputResponse, index) => {
         const bundle = mapRareSatsAPIResponseToBundle(inputResponse);
-        value += bundle.value;
-        totalExoticSats += bundle.totalExoticSats;
-        satRanges.push(...bundle.satRanges);
+        if (
+          bundle.inscriptions.length > 0 ||
+          bundle.satributes.some((satributes) => !satributes.includes('COMMON'))
```
Comment (teebszet) [inline src/app/hooks/useDetectOrdinalInSignPsbt.ts:31]: what if `bundle.satributes: ['COMMON', 'UNCOMMON']`

it won't get pushed into bundleItemsData?

File: src/app/hooks/useDetectOrdinalInSignPsbt.ts [hunk 3]
```diff
@@ -1,33 +1,41 @@
...
 const useDetectOrdinalInSignPsbt = () => {
   const { ordinalsAddress, network } = useWalletSelector();
 
-  const handleOrdinalAndOrdinalInfo = async (parsedPsbt?: ParsedPSBT) => {
-    const satRanges: BundleSatRange[] = [];
-    let value = 0;
-    let totalExoticSats = 0;
+  const handleOrdinalAndOrdinalInfo = async (
+    parsedPsbt?: ParsedPSBT,
+  ): Promise<{ bundleItemsData: InputsBundle; userReceivesOrdinal: boolean }> => {
```
Comment (teebszet) [inline src/app/hooks/useDetectOrdinalInSignPsbt.ts:18]: 👏 I guess we will move this into core, once you have to implement for mobile?

File: src/app/screens/signBatchPsbtRequest/index.tsx [hunk 4]
```diff
@@ -366,14 +367,32 @@ function SignBatchPsbtRequest() {
 
                 {userTransfersOrdinals.length > 0 &&
                   userTransfersOrdinals.map((item, index) => (
-                    // eslint-disable-next-line react/no-array-index-key
-                    <SatsBundle key={`${index}-transfer`} bundle={item as Bundle} />
+                    <SatsBundle
+                      // eslint-disable-next-line react/no-array-index-key
+                      key={`${index}-transfer`}
+                      title={`${tCommon('INPUT')} #${item.inputIndex + 1}`}
```
Comment (teebszet) [inline src/app/screens/signBatchPsbtRequest/index.tsx:373]: check with @victorkirov and @rektbap on slack, might be 0-based index here

**Review by teebszet — APPROVED** [review]

**Pull request merged**